psydac/api/ast/nodes.py

```diff
@@ -1380,6 +1410,30 @@ def set_index(self, index):
         return GlobalSpan(self.target, index)
 
 #==============================================================================
+class LocalSpan(ArrayNode):
+    """
+     This represents the local span array
+    """
+    _rank = 1
+    _positions = {index_element: 0}
+
+    def __new__(cls, target, index=None):
+        if not isinstance(target, (ScalarFunction, VectorFunction, IndexedVectorFunction)):
+            raise TypeError('Expecting a scalar/vector test function')
+
+        return Basic.__new__(cls, target, index)
+
+    @property
+    def target(self):
+        return self._args[0]
+
+    @property
+    def index(self):
+        return self._args[1]
+
+    def set_index(self, index):
+        return LocalSpan(self.target, index)
```

Comment:
Actually I have just realized that there is already a `Span` class in this file, which is almost identical to `GlobalSpan` and `LocalSpan`. The only difference is that it subclasses `ScalarNode` instead of `ArrayNode`.

I think that we should clean this up.

Reply:
yes, we have one for `Span` which represents a scalar, the others are for arrays that contain the spans

Reply:
I suggest to use new names:

- `Span(ScalarNode)` stays the same
- `SpanArray(ArrayNode)` will have the methods `__new__`, `target`, `index` and`set_index` as I showed above
- `GlobalSpanArray(SpanArray)` will only contain a docstring as above
- `LocalSpanArray(SpanArray)` will only contain a docstring as above

Reply:
okay, I will change it

Reply:
done in [654cb3e](https://github.com/pyccel/psydac/pull/213/commits/654cb3e44ddd927ab80cde1e8c7d5614f6dc92bb)